src/syntax/parser.rs

```diff
@@ -1,449 +1,212 @@
-#![allow(unused_parens)]
+extern crate sqlparser;
 
-use engine::aggregator::Aggregator;
-use engine::query::*;
-use ingest::raw_val::RawVal;
-use nom::{digit, is_alphabetic, is_alphanumeric, multispace};
-use std::boxed::Box;
-use std::str;
-use std::str::FromStr;
+use sqlparser::sqlparser::*;
+use sqlparser::sqlast::*;
+use engine::query::Query;
 use syntax::expression::*;
-use syntax::limit::LimitClause;
-use time;
-
-
-named!(pub parse_query<&[u8], Query>, alt_complete!(full_query | simple_query));
-
-named!(full_query<&[u8], Query>,
-    do_parse!(
-        opt!(multispace) >>
-        tag_no_case!("select") >>
-        multispace >>
-        select: select_clauses >>
-        opt!(multispace) >>
-        table: from_clause >>
-        multispace >>
-        tag_no_case!("where") >>
-        multispace >>
-        filter: expr >>
-        opt!(multispace) >>
-        order_by: opt!(order_by_clause) >>
-        opt!(multispace) >>
-        limit: opt!(limit_clause) >>
-        opt!(multispace) >>
-        char!(';') >>
-        (construct_query(select, table, filter, order_by, limit))
-    )
-);
-
-
-named!(simple_query<&[u8], Query>,
-    do_parse!(
-        opt!(multispace) >>
-        tag_no_case!("select") >>
-        multispace >>
-        select: select_clauses >>
-        opt!(multispace) >>
-        table: from_clause >>
-        opt!(multispace) >>
-        order_by: opt!(order_by_clause) >>
-        opt!(multispace) >>
-        limit: opt!(limit_clause) >>
-        opt!(multispace) >>
-        opt!(char!(';')) >>
-        (construct_query(select, table, Expr::Const(RawVal::Int(1)), order_by, limit))
-    )
-);
+use engine::aggregator::*;
+use ingest::raw_val::RawVal;
+use syntax::limit::*;
+use sqlparser::dialect::GenericSqlDialect;
+use QueryError;
+
+// Convert sqlparser-rs `ASTNode` to LocustDB's `Query`
+pub fn parse_query(query: &str) -> Result<Query, QueryError> {
+    let dialect = GenericSqlDialect {};
+    let ast = match Parser::parse_sql(&dialect, query.to_string()) {
+        Ok(ast) => ast,
+        Err(e) => {
+           if let ParserError::ParserError(e_str) = e {
+               return Err(QueryError::ParseError(e_str));
+           }
+           return Err(QueryError::FatalError(format!("{:?}", e)));
+        }
+    };
+
+    let mut select = Vec::<Expr>::new();
+    let mut aggregate = Vec::<(Aggregator, Expr)>::new();
+    let mut table = String::new();
+    let mut filter = Expr::Const(RawVal::Int(1));
+    let mut order_by_str = None;
+    let mut order_desc = false;
+    let mut limit_clause = LimitClause { limit: 100, offset: 0 };
+    let order_by_index = None;
+
+    if let ASTNode::SQLSelect 
```

Comment:
if `ast` is something other than select, we'd want to return error rather than empty query
code can probably be simplified (and a bunch of `mut`s removed) by always returning error when ast can't be fully translated